editor/blocks/quote/index.js

```diff
@@ -0,0 +1,42 @@
+const Editable = wp.blocks.Editable;
+const { html } = wp.blocks.query;
+
+wp.blocks.registerBlock( 'core/quote', {
+	title: 'Quote',
+	icon: 'format-quote',
+	category: 'common',
+
+	attributes: {
+		value: html( 'blockquote > p' ),
+		citation: html( 'footer' )
+	},
+
+	edit( attributes, onChange ) {
+		const { value, citation } = attributes;
+
+		return (
+			<blockquote>
+				<Editable
+					value={ value }
+					onChange={ ( newValue ) => onChange( { value: newValue } ) } />
+				<footer>
+					<Editable
+						value={ citation }
+						onChange={ ( newValue ) => onChange( { citation: newValue } ) } />
+				</footer>
+			</blockquote>
+		);
+	},
+
+	save( attributes ) {
+		const { value, citation } = attributes;
+		return (
+			<blockquote>
+				{ value }
+				<footer>
```

Comment:
We may want to account for the case that `citation` is empty and avoid including the `footer` if possible. This was one of the reasons for `hpq`: to account for cases where the markup isn't always necessarily the same in the generated output (maybe a quote citation, maybe an image caption, but not necessarily).